### DISCUSSION TITLE
ensure "time" is part of the specification

### DISCUSSION THREAD

**Discussion by ljharb** [npm/feedback#260] [Registry] [answered]
https://twitter.com/ghchangelog/status/1367900801184710658?s=21 implies that the "time" data (ie, `npm show npm time`) isn't part of the "official npm specification".

It should be, since it's *absolutely critical* when debugging to be able to know what date/time all versions were published. It's the only way to figure out what I should put into  `--before`.

(if i'm misunderstanding, and `npm show $spec time` would continue to work for all compliant registries, then this can be closed)

**Comment by MylesBorins** [comment]
@ljharb is this request related to GitHub packages? Is your request to ensure that 3rd party registries need to implement this to be considered "compliant"?

To the best of my knowledge there isn't a spec that 3rd party registries are working against, so we would need to have something like that to even be able to enforce anything.

**Reply by targos** [reply]
If there is no spec, what do you think "the official npm specification" means in https://github.blog/changelog/2021-03-05-packages-time-has-been-dropped-from-npm-package-metadata-responses/ ?

**Reply by MylesBorins** [reply]
TBH that is the first I've heard of this. It is highly possible I'm just out of the loop. I'll dig in a report back

**Reply by ljharb** [reply]
That’s exactly what I’m responding to. It appears that GitHub packages has intentionally broken a critical workflow (one i don’t currently rely on for that registry, but do for others) on the basis that it’s not part of some npm specification - I’d like to ensure that, if it exists, such a specification does mandate this critical workflow be achievable, so that when i have occasion to use the GitHub packages registry, I’m not stuck.

**Reply by MylesBorins** [reply]
I'm following up with folks internally to figure out more about the "spec"

**Reply by lumaxis** [reply]
Could this possibly be referring to https://docs.npmjs.com/cli/v7/configuring-npm/package-json?

**Comment by MylesBorins** [comment, marked as answer]
Quick update for y'all.

I've dug in internally and confirmed that there currently is NOT an official "npm spec". Also the removal of `time` was due to performance reasons. We continue to have all the data necessary to return a `time` value as part of the metadata response and will resume returning this value in the future once we have solved the existing performance issues.

https://github.blog/changelog/2021-03-05-packages-time-has-been-dropped-from-npm-package-metadata-responses/
